include/verilated_trace.h

```diff
@@ -26,6 +26,8 @@
 
 #include "verilated.h"
 
+#include "gtkwave/fstapi.h"
+
```

Comment:
This will require gtkwave/fstapi.h even when not using FST traces, so please don't include that here.

Perhaps add a virtual function to print the scope type and overload in the FST class, or otherwise refactor, thanks.

Reply:
I did it this way to be able to use the `FST_ST_VCD_*` in `verilated_vcd_c`, since it is defined in `gtkwave/fstapi.h`. Right now Verilator will generate code using the values defined by that enum both for FST and VCD traces.

 The values could also be hard-coded in  `verilated_vcd_c.cpp` or a smaller enum could be created with just the required identifiers in `verilated_vcd_c.h`.

What do you think?

Reply:
Hmm.
1. We could make our own enum.
2. We could include a new fstdefs.h, which separates the fstapi.h definitions from its exports.  This might be cleanest. 
3. We could include fstapi.h, but as fstapi.h includes things like zlib.h that isn't on all platforms.  I tried removing the include of zlib.h and time.h as both seemed unneeded in fstapi.h and I put them into fstapi.cpp and this seemed to work ok. That move should probably be fed upstream regardless of this issue, as no need to have those included in the header.

What do you think?

Perhaps feel out with gtkwave project if they are willing to separate out the enum definitions?

Reply:
If we do 1. we should add some asserts in e.g. verilated_fst_c.cpp to make sure the new verilator enum values match FST_ST_VCD_*.

Reply:
Doing 1. with assertions should be good enough. Half of `fstScopeType` is for VHDL, leaving us only with 10 or so items, which seems manageable.

The include at V3TraceDecl.cpp could be changed to include the new enum, so we should either create a new `tracedefs.h` file or include `verilated_trace.h`.